### PR TITLE
updating oneDAL roadmap

### PR DESCRIPTION
**Pull request by napetrov**
Updating oneDAL roadmap as we will focus more on overall structure definition and not new algorithms definition

### CODE REVIEW

**Comment by rscohn2** [comment]
@rlnx @andreynv  Should this be merged?

**Comment by napetrov** [comment]
@rdower  - we sync offline with @rlnx @andreynv   and i will update this slightly now

**Comment by napetrov** [comment]
@andreynv  - API list was also fixed for 0.8

File: roadmap.rst [hunk 1]
```diff
@@ -159,7 +159,7 @@ Date       Milestone
   - Add overall spec skeleton, introduction and algorithm classes sections
   - Add initial version for sections on data structures (descriptors algorithms/tables, numeric tables),
     compute modes(batch/online), distributed algorithms and generic building blocks(train/predict) 
-  - Add API for 3 oneDAL algorithms
```
Comment (andreynv) [inline roadmap.rst:162]: We mention kNN  somewhere below; why to mention it again

Comment (napetrov) [reply roadmap.rst]: fixed

File: roadmap.rst [hunk 2]
```diff
@@ -225,8 +225,8 @@ Date       Milestone
   
 - oneDAL
 
-  - Extend API description to 18 algorithms
-  - Add first version for sections on programming model, algorithmics
+  - Extend API description to kNN classification
+  - Add first version for sections on error handling, namespaces, terminology, math notations
```
Comment (andreynv) [inline roadmap.rst:229]: We can be flexible and do not tell about it, but as I understand we describe basic things related to data management (tables/accessors/builders) andd algorithms (descriptors, train etc)

**Pull request merged**